superset/db_engine_specs.py

```diff
@@ -111,8 +111,10 @@ class BaseEngineSpec(object):
     time_secondary_columns = False
     inner_joins = True
     allows_subquery = True
+    supports_column_aliases = True
```

Comment:
It would be good to add a comment on what this means, or just for our own understanding: It means not only that the DB supports the "AS" keyword but also it actually returns the columns with the specified alias name ... ie alias is supported both syntactically and semantically. 

For example in Pinot, it does support the "AS" keyword syntactically but its not actually returned with it.

Interestingly as @betodealmeida points out, this is also violated at times by postgres: it returns the returned column name as 'sum(foo)' instead of 'SUM(foo)' sometimes. 

So should we say that even for PG supports_column_aliases should be false ?

Reply:
I think it's good to limit the use of `supports_column_aliases` to engines that truly don't support aliases. Even though the new mutation logic makes the aliases less important, they still serve two purposes:
1. Rendering a more descriptive query in the `View Query` tab that can be copied and used elsewhere.
2. Making the subqueries in more complex queries work. In some cases a column from a subquery can e.g. exceed the max column name length, in which case it needs to be mutated.

It is my understanding that Postgres requires quotes around case-sensitive alises. Eg. `select 1 as x_Col` will return the header `x_col`, while `select 1 as "x_Col"` will return `x_Col`. If I remember correctly `psycopg2` does this quoting automatically, but if not, it might make sense to set `force_column_alias_quotes` to True. I can test this more thoroughly.

Reply:
I tested this on Postgres and wasn't able to reproduce case sensitive cols not being automatically quoted. See below how `lowercol`(=lowercase) is not quoted but `mixedCol`(=mixed case) is automatically quoted in the main query.
<img width="894" alt="screenshot 2019-02-11 at 17 40 51" src="https://user-images.githubusercontent.com/33317356/52587990-1666cb00-2e44-11e9-93bc-7990e130a950.png">